<!--
Authors : Siqing MA
Date: 2018-7-28 created, 2018-9-16 updated
Multi turtlebot3 follow start
-->

<launch>

    <arg name="tb3_0_disperse_point_x" default=" -3.0"/>
    <arg name="tb3_0_disperse_point_y" default=" -5.0"/>
    
    <arg name="tb3_1_disperse_point_x" default=" -1.0"/>
    <arg name="tb3_1_disperse_point_y" default=" 0.0"/>

    <arg name="tb3_2_disperse_point_x" default=" 1.0"/>
    <arg name="tb3_2_disperse_point_y" default=" 1.0"/>

    <node name="turtle1_tf_broadcaster" pkg="multi_turtlebot_opt" type="turtle_tf_broadcaster.py" respawn="false" output="screen" >
      <param name="turtle" type="string" value="tb3_0" />
    </node>

    <node name="turtle2_tf_broadcaster" pkg="multi_turtlebot_opt" type="turtle_tf_broadcaster.py" respawn="false" output="screen" >
      <param name="turtle" type="string" value="tb3_1" /> 
    </node>

    <node name="turtle3_tf_broadcaster" pkg="multi_turtlebot_opt" type="turtle_tf_broadcaster.py" respawn="false" output="screen" >
      <param name="turtle" type="string" value="tb3_2" /> 
    </node>

   
    <node name="tb3_0_goal_tf_broadcaster" pkg="multi_turtlebot_opt" type="goal_tf_broadcaster.py" respawn="false" output="screen" >
        <param name="goal_id" type="string" value="tb3_0_goal" />
	<param name="goal_x"  value="$(arg tb3_0_disperse_point_x)" />
	<param name="goal_y"  value="$(arg tb3_0_disperse_point_y)" />
    </node>

    <node name="tb3_1_goal_tf_broadcaster" pkg="multi_turtlebot_opt" type="goal_tf_broadcaster.py" respawn="false" output="screen" >
      	<param name="goal_id" type="string" value="tb3_1_goal" />
      	<param name="goal_x"  value="$(arg tb3_1_disperse_point_x)" />
	<param name="goal_y"  value="$(arg tb3_1_disperse_point_y)" /> 
    </node>

    <node name="tb3_2_goal_tf_broadcaster" pkg="multi_turtlebot_opt" type="goal_tf_broadcaster.py" respawn="false" output="screen" >
      	<param name="goal_id" type="string" value="tb3_2_goal" />
	<param name="goal_x"  value="$(arg tb3_2_disperse_point_x)" />
	<param name="goal_y"  value="$(arg tb3_2_disperse_point_y)" /> 
    </node>


    <node pkg="multi_turtlebot_opt" type="DWA_control.py" name="follower_controls_0" output="screen">
	<param name="mode" type="string" value="disperse" /> 
      	<param name="tb3_id" type="string" value="tb3_0" /> 
      	<param name="goal_id" type="string" value="tb3_0_goal" /> 
    </node>

    <node pkg="multi_turtlebot_opt" type="DWA_control.py" name="follower_controls_1" output="screen">
	<param name="mode" type="string" value="disperse" /> 
      	<param name="tb3_id" type="string" value="tb3_1" /> 
      	<param name="goal_id" type="string" value="tb3_1_goal" /> 
    </node>

    <node pkg="multi_turtlebot_opt" type="DWA_control.py" name="follower_controls_2" output="screen">
	<param name="mode" type="string" value="disperse" /> 
      	<param name="tb3_id" type="string" value="tb3_2" /> 
      	<param name="goal_id" type="string" value="tb3_2_goal" /> 
    </node>

</launch>
